src/commonMain/kotlin/fr/acinq/lightning/channel/InteractiveTx.kt

```diff
@@ -48,27 +53,28 @@ sealed class SharedFundingInput {
 data class InteractiveTxParams(
     val channelId: ByteVector32,
     val isInitiator: Boolean,
-    val localAmount: Satoshi,
-    val remoteAmount: Satoshi,
+    val localContribution: Satoshi,
+    val remoteContribution: Satoshi,
     val sharedInput: SharedFundingInput?,
     val fundingPubkeyScript: ByteVector,
     val localOutputs: List<TxOut>,
     val lockTime: Long,
     val dustLimit: Satoshi,
     val targetFeerate: FeeratePerKw
 ) {
-    constructor(channelId: ByteVector32, isInitiator: Boolean, localAmount: Satoshi, remoteAmount: Satoshi, fundingPubkeyScript: ByteVector, lockTime: Long, dustLimit: Satoshi, targetFeerate: FeeratePerKw) :
-            this(channelId, isInitiator, localAmount, remoteAmount, null, fundingPubkeyScript, listOf(), lockTime, dustLimit, targetFeerate)
+    constructor(channelId: ByteVector32, isInitiator: Boolean, localContribution: Satoshi, remoteContribution: Satoshi, fundingPubkeyScript: ByteVector, lockTime: Long, dustLimit: Satoshi, targetFeerate: FeeratePerKw) :
+            this(channelId, isInitiator, localContribution, remoteContribution, null, fundingPubkeyScript, listOf(), lockTime, dustLimit, targetFeerate)
 
-    init {
-        require(localAmount >= 0.sat && remoteAmount >= 0.sat) { "funding amount cannot be negative" }
-    }
-
-    val fundingAmount: Satoshi = localAmount + remoteAmount
     // BOLT 2: MUST set `feerate` greater than or equal to 25/24 times the `feerate` of the previously constructed transaction, rounded down.
     val minNextFeerate: FeeratePerKw = targetFeerate * 25 / 24
     // BOLT 2: the initiator's serial IDs MUST use even values and the non-initiator odd values.
     val serialIdParity = if (isInitiator) 0 else 1
+
+    /**
+     * @param previousFundingAmount if we're splicing over an existing channel, amount of the previous funding output.
+     * @return the amount of the new funding output.
+     */
+    fun fundingAmount(previousFundingAmount: Satoshi): Satoshi = previousFundingAmount + localContribution + remoteContribution
```

Comment:
😎 

```suggestion

       /**
     * @return the amount of the new funding output, which the sum of the shared input, if any, and both side's contributions.
     */
    fun fundingAmount: Satoshi = (sharedInput?.info?.txOut?.amount ?: 0.sat) + localContribution + remoteContribution
```

Reply:
Nice, good catch, I found it annoying to make this a function! Fixed in https://github.com/ACINQ/lightning-kmp/pull/428/commits/af7e7ef33cc2ecad04f24f83b1367722e68a514c